toolkit/tools/pkg/imagecustomizerlib/liveosisobuilder.go

```diff
@@ -0,0 +1,671 @@
+//     `LiveOSIsoBuilder.workingDirs.outDir` folder.
+//   - the paths to individual artifaces are found in the
+//     `LiveOSIsoBuilder.artifacts` data structure.
+func (b *LiveOSIsoBuilder) prepareArtifactsFromFullImage(rawImageFile string) error {
+
+	logger.Log.Infof("Connecting to raw image (%s)", rawImageFile)
+	imageConnection, mountPoints, err := connectToExistingImage(rawImageFile, b.workingDirs.isoBuildDir, "imageroot", true)
+	if err != nil {
+		return err
+	}
+	defer imageConnection.CleanClose()
+
+	bootMountPoint := safechroot.FindMountPointByTarget(mountPoints, "/boot/efi")
+	if bootMountPoint == nil {
+		return fmt.Errorf("failed to find boot partition mount point in %s", rawImageFile)
+	}
+
+	err = b.extractArtifactsFromBootDevice(bootMountPoint.GetSource(), bootMountPoint.GetFSType())
+	if err != nil {
+		return fmt.Errorf("failed to extract boot artifacts from image (%s).\n%w", rawImageFile, err)
+	}
+
+	rootfsMountPoint := safechroot.FindMountPointByTarget(mountPoints, "/")
+	if rootfsMountPoint == nil {
+		return fmt.Errorf("failed to find rootfs partition mount point in %s", rawImageFile)
+	}
+
+	writeableRootfsDir := filepath.Join(b.workingDirs.isoBuildDir, "writeable-rootfs")
+	err = b.populateWriteableRootfsDir(rootfsMountPoint.GetSource(), rootfsMountPoint.GetFSType(), writeableRootfsDir)
+	if err != nil {
+		return fmt.Errorf("failed to copy the contents of rootfs from image (%s) to local folder (%s).\n%w", rawImageFile, writeableRootfsDir, err)
+	}
+
+	isoMakerArtifactsStagingDir := "/boot-staging"
+	err = b.prepareLiveOSDir(writeableRootfsDir, isoMakerArtifactsStagingDir)
+	if err != nil {
+		return fmt.Errorf("failed to convert rootfs folder to a LiveOS folder.\n%w", err)
+	}
+
+	err = b.createSquashfsImage(writeableRootfsDir)
+	if err != nil {
+		return fmt.Errorf("failed to create squashfs image.\n%w", err)
+	}
+
+	isoMakerArtifactsDirInInitrd := "/boot"
+	err = b.generateInitrdImage(writeableRootfsDir, isoMakerArtifactsStagingDir, isoMakerArtifactsDirInInitrd)
+	if err != nil {
+		return fmt.Errorf("failed to generate initrd image.\n%w", err)
+	}
+
+	err = os.RemoveAll(writeableRootfsDir)
+	if err != nil {
+		return fmt.Errorf("failed to remove working folder (%s).\n%w", writeableRootfsDir, err)
+	}
+
+	return nil
+}
+
+// createIsoMakerConfig
+//
+//	creates an IsoMaker config objects with the necessary configuration.
+//
+// inputs:
+//   - squashfsImagePath [in]:
+//   - path to an existing squashfs image file. The configuration will instruct
+//     IsoMaker to place it under:
+//   - /config/additionalfiles/0/$(basename $squashfsImagePath).
+//
+// outputs:
+//   - returns an IsoMaker configuration.Config object.
+func createIsoMakerConfig(squashfsImagePath string) (configuration.Config, error) {
+
+	config := configuration.Config{
+		SystemConfigs: []configuration.SystemConfig{
+			{
+				AdditionalFiles: map[string]configuration.FileConfigList{
+					// 'AdditionalFiles' is meant to do two things:
```

Comment:
IsoMaker assumes in a lot of places the config/iso structure of the Mariner iso build.
I thought about doing more refactoring in IsoMaker - but I wanted to do that separately and also only as needed to avoid disturbing the code too much at once and folding more working into these mic changes.
So, yep, I agree we should make IsoMaker more generic. I propose to do that separately and wait and see how it plays with the other scenarios we want first.